lib/urlapi.c

```diff
@@ -744,6 +752,24 @@ static CURLUcode seturl(const char *url, CURLU *u, unsigned int flags)
     memcpy(hostname, hostp, len);
     hostname[len] = 0;
 
+    if((flags & CURLU_GUESS_SCHEME) && !schemep) {
+      /* legacy curl-style guess based on host name */
+      if(checkprefix("FTP.", hostname))
+        schemep = (char *)"ftp";
+      else if(checkprefix("DICT.", hostname))
+        schemep = (char *)"DICT";
+      else if(checkprefix("LDAP.", hostname))
+        schemep = (char *)"LDAP";
+      else if(checkprefix("IMAP.", hostname))
+        schemep = (char *)"IMAP";
```

Comment:
Why are some protocols returned in uppercase when the `protocols` array in `lib/version.c` lists all in lowercase?

Reply:
oops, I want them lowercase!